bundles/org.openhab.binding.mqtt.generic/src/main/java/org/openhab/binding/mqtt/generic/values/PercentageValue.java

```diff
@@ -105,8 +105,8 @@ public PercentType parseCommand(Command command) throws IllegalArgumentException
             }
         } else //
                // On/Off equals 100 or 0 percent
-        if (command instanceof OnOffType increaseDecreaseCommand) {
-            return increaseDecreaseCommand == OnOffType.ON ? PercentType.HUNDRED : PercentType.ZERO;
+        if (command instanceof OnOffType onOffCommand) {
```

Comment:
Unused template variable `onOffCommand`. You can remove it or use as return value.